front/src/modules/rollingStock/components/rollingStockEditor/RollingStockEditorCurves.tsx

```diff
@@ -197,44 +244,86 @@ export default function RollingStockEditorCurves({
   const updateCurrentRs = (e: Matrix<{ value: string }>) => {
     const parsedCurve = parseCurve(e);
     if (!isEmpty(parsedCurve.max_efforts) && !isEmpty(parsedCurve.speeds)) {
-      // ** TODO: Add update CurrentRsEffortCurve **
+      /* * * 1st find the index of the modified curve in the curves list, * *
+       * * then slice the list to replace the selected curve by the new one */
+      const index = currentRsEffortCurve.modes[selectedTractionMode].curves.findIndex(
+        (curve) =>
+          curve.cond?.comfort === selectedComfortLvl &&
+          curve.cond?.electrical_profile_level === selectedElectricalProfile &&
+          curve.cond?.power_restriction_code === selectedPowerRestriction
+      );
+
+      const updatedSelectedCurve: ConditionalEffortCurve = {
+        cond: currentRsEffortCurve.modes[selectedTractionMode].curves[index].cond,
+        curve: parsedCurve,
+      };
 
-      setSelectedCurve((prevState) => ({
+      const updatedCurvesList = currentRsEffortCurve.modes[selectedTractionMode].curves
+        .slice(0, index)
+        .concat(
+          updatedSelectedCurve,
+          currentRsEffortCurve.modes[selectedTractionMode].curves.slice(index + 1)
+        );
```

Comment:
You can use splice here :)
```suggestion
      const updatedCurvesList = selectedTractionModeCurves.splice(index, 1, updatedSelectedCurve);
```